flocker/node/test/test_deploy.py

```diff
@@ -1312,18 +1319,22 @@ def test_application_needs_starting(self):
 
         nodes = frozenset([
             Node(
-                hostname=u'example.com',
+                uuid=api.node_uuid,
```

Comment:
This and the other similar changes to avoid hitting the deprecated codepaths are good ... but would be a lot easier to review in a separate PR (particularly one that makes non-test changes).